ui/src/layouts/storage.rs

```diff
@@ -342,42 +375,91 @@ pub fn FilesLayout(cx: Scope<Props>) -> Element {
                         rsx!(
                         Folder {
                             with_rename: true,
-                            onrename: |val| {
+                            onrename: |(val, key_code)| {
                                 let new_name: String = val;
-                                if STATIC_ARGS.use_mock {
-                                    directories_list
-                                        .with_mut(|i| i.insert(0, Directory::new(&new_name)));
-                                        update_items_with_mock_data(
-                                            storage_state,
-                                            current_dir,
-                                            dirs_opened_ref,
-                                            directories_list,
-                                            files_list,
-                                        );
-                                } else {
-                                    ch.send(ChanCmd::CreateNewDirectory(new_name));
-                                    ch.send(ChanCmd::GetItemsFromCurrentDirectory);
+                                if key_code == Code::Enter {
+                                    if STATIC_ARGS.use_mock {
+                                        directories_list
+                                            .with_mut(|i| i.insert(0, Directory::new(&new_name)));
+                                            update_items_with_mock_data(
+                                                storage_state,
+                                                current_dir,
+                                                dirs_opened_ref,
+                                                directories_list,
+                                                files_list,
+                                            );
+                                    } else {
+                                        ch.send(ChanCmd::CreateNewDirectory(new_name));
+                                        ch.send(ChanCmd::GetItemsFromCurrentDirectory);
+                                    }
                                 }
                                 add_new_folder.set(false);
                              }
                         })
                     }),
                     directories_list.read().iter().map(|dir| {
                         let folder_name = dir.name();
-                        rsx!(Folder {
-                            text: dir.name(),
-                            aria_label: dir.name(),
-                            onpress: move |_| {
-                                ch.send(ChanCmd::OpenDirectory(folder_name.clone()));
-                            }
-                        })
+                        let folder_name2 = dir.name();
+                        let is_renaming = use_state(cx, || false);
```

Comment:
calling `use_state` or any other hook inside of a loop, iterator, or conditional statement is bad. don't ever do this. if the files list changes,  then one invocation of `FilesLayout` will produce a different set of hooks then the next. 

This violates the [rules of hooks](https://dioxuslabs.com/docs/0.3/guide/en/interactivity/hooks.html?highlight=hooks#rules-of-hooks). 

instead of using a `ContextMenu` you need to make a new element which contains a `ContextItem` and a `File`, create the hook in the top level element, and pass it to both child elements.

Reply:
sorry if i sounded overly bossy in the first paragraph.